cinder-{{cookiecutter.driver_name_lc}}/src/charm.py

```diff
@@ -15,75 +15,34 @@
 
 
-class Cinder{{ cookiecutter.driver_name }}Charm(OSBaseCharm):
+class CinderCharmBase(CinderStoragePluginCharm):
 
-    _stored = StoredState()
     PACKAGES = ['{{ cookiecutter.additional_package_name|default("cinder-common", true) }}']
+    # Overriden from the parent. May be set depending on the charm's properties
+    stateless = {{ cookiecutter.stateless }}
+    active_active = {{ cookiecutter.active_active }}
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        # We listen to the following events:
-        # - Installation: Just to set the charm status
-        # - Configuration changes: Rewrite the cinder.conf file and inform
-        #   other charms of the changes
-        # - Backend storage join/change: Inform the other charm of what the
-        #   configuration is and the backend name.
-        self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.config_changed, self._on_config)
-        self.framework.observe(
-            self.on.storage_backend_relation_joined,
-            self._on_storage_backend)
-        self.framework.observe(
-            self.on.storage_backend_relation_changed,
-            self._on_storage_backend)
 
-    def _on_install(self, _):
-        self.install_pkgs()
-        self.unit.status = ActiveStatus('Unit is ready')
-
-    def _render_config(self, config, app_name):
-        # Generate the JSON with the updated configuration.
+    def cinder_configuration(self, config):
+        # Return the configuration to be set by the principal.
         volume_driver = ''
         options = [
-            ('volume_driver', volume_driver),
+            ('volume_driver', volume_driver)
         ]
-        return json.dumps({
-            "cinder": {
-                "/etc/cinder/cinder.conf": {
-                    "sections": {app_name: options}
-                }
-            }
-        })
-
-    def _set_data(self, data, config, app_name):
-        # Inform another charm of the backend name and our configuration.
-        data['backend-name'] = config['volume-backend-name'] or app_name
-        data['subordinate_configuration'] = self._render_config(
-            config, app_name)
+        return options
 
-    def _on_config(self, event):
-        config = dict(self.framework.model.config)
-        rel = self.framework.model.relations.get('storage-backend')[0]
-        app_name = self.framework.model.app.name
-        for unit in self.framework.model.get_relation('storage-backend').units:
-            self._set_data(rel.data[self.unit], config, app_name)
-        self.unit.status = ActiveStatus('Unit is ready')
 
-    def _on_storage_backend(self, event):
-        self._set_data(
-            event.relation.data[self.unit],
-            self.framework.model.config,
-            self.framework.model.app.name)
+@charm_class
+class Cinder{{ cookiecutter.driver_name }}Charm(CinderCharmBase):
+    release = 'ussuri'
```

Comment:
Isn't this to be `cookiecutter.release`?

Reply:
Yes. Good catch.